pkg/common/morpc/codec_header.go

```diff
@@ -50,3 +51,43 @@ func (hc *deadlineContextCodec) Decode(msg *RPCMessage, data []byte) (int, error
 	msg.Ctx, msg.cancel = context.WithTimeout(msg.Ctx, time.Duration(buf.Byte2Int64(data)))
 	return 8, nil
 }
+
+type traceCodec struct {
+}
+
+func (hc *traceCodec) Encode(msg *RPCMessage, out *buf.ByteBuf) (int, error) {
+	if msg.Ctx == nil {
+		return 0, nil
+	}
+
+	span := trace.SpanFromContext(msg.Ctx)
+	c := span.SpanContext()
+	n := c.Size()
+	out.MustWriteByte(byte(n))
```

Comment:
encode size, is it necessary? it can get size from SpanContext

Reply:
Yes. There are many headers in the buf. Size is used to decode trace context